frontend/public/components/environment.jsx

```diff
@@ -44,212 +45,214 @@ export const envVarsToArray = (initialPairObjects) => {
   return [getPairsFromObject(initialPairObjects)];
 };
 
-export class EnvironmentPage extends PromiseComponent {
+const stateToProps = ({k8s}, {obj}) => ({
+  model: k8s.getIn(['RESOURCES', 'models', referenceFor(obj)]) || k8s.getIn(['RESOURCES', 'models', obj.kind]),
+});
+
+export const EnvironmentPage = connect(stateToProps)(
+  class EnvironmentPage extends PromiseComponent {
   /**
    * Set initial state and decide which kind of env we are setting up
    *
    * @param props
    */
-  constructor(props) {
-    super(props);
+    constructor(props) {
+      super(props);
 
-    this.reload = () => this._reload();
-    this.saveChanges = (...args) => this._saveChanges(...args);
-    this.updateEnvVars = (...args) => this._updateEnvVars(...args);
+      this.reload = () => this._reload();
+      this.saveChanges = (...args) => this._saveChanges(...args);
```

Comment:
Just a side comment, this is a bit different than the recommended `this.someFunc = this.someFunc.bind(this);`.

Reply:
but after that commented added the `w=1` to eliminate the whitespace changes, this PR doesn't cover the above.  :)

Reply:
@benjaminapetersen thats ok, I'll make the change anyway! Thanks for the review.